app/tests/unit/services/object.spec.js

```diff
@@ -0,0 +1,139 @@
+const { MockModel, MockTransaction } = require('../../common/dbHelper');
+
+jest.mock('../../../src/db/models/tables/objectModel', () => MockModel);
+
+const service = require('../../../src/services/object');
+const objectPermissionService = require('../../../src/services/objectPermission');
+
+const { Permissions } = require('../../../src/components/constants');
+
+const bucketId = '00000000-0000-0000-0000-000000000000';
+const objectId = '00000000-0000-0000-0000-000000000123';
+const userId = '00000000-0000-0000-0000-000000000000';
+
+const data = {
+  id: objectId,
+  bucketId: bucketId,
+  path: 'path',
+  public: 'true',
+  active: 'true',
+  createdBy: userId,
+  userId: userId
+};
+
+const params = {
+  bucketId: bucketId,
+  bucketName: 'bucketName',
+  active: 'true',
+  key: 'key',
+  userId: userId
+};
+
+beforeEach(() => {
+  MockModel.mockReset();
+  MockTransaction.mockReset();
+});
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('create', () => {
+
+  it('should add object permissions if user exists', async () => {
+
+    // add userId property
+    const dataWithUser = { ...data, userId: userId };
+
+    // create perms object
+    const perms = Object.values(Permissions).map((p) => ({
+      userId: dataWithUser.userId,
+      permCode: p
+    }));
+
+    // spy on function
+    const addPermissionSpy = jest.spyOn(objectPermissionService, 'addPermissions');
+
+    // mocking the stuff we dont want to test
+    const etrx = await jest.fn().mockResolvedValue(MockTransaction);
+    addPermissionSpy.mockResolvedValue({});
+
+    // call the function we're testing
+    await service.create(dataWithUser, etrx);
+
+    // the tests:
+    // userId was passed, so addPermissions was called
+    expect(addPermissionSpy).toHaveBeenCalledTimes(1);
+    // addPermissions was called with expected paramters
+    expect(addPermissionSpy).toHaveBeenCalledWith(objectId, perms, userId, etrx);
+    // expect(MockModel.query).toHaveBeenCalledTimes(1);
+  });
+
+  it.skip('should add object permissions, no transaction provided', async () => {
+
+    // add userId property
+    const dataWithUser = { ...data, userId: userId };
```

Comment:
we dont need to add the userId, that was for the previous test. can delete this line.
and call `await service.create(data);` instead below